schema.yaml

```diff
@@ -0,0 +1,257 @@
+"$schema": https://json-schema.org/draft/2020-12/schema#
+title: CDBA Configuration Schema
+type: object
+properties:
+  devices:
+    type: array
+    items:
+      type: object
+      properties:
+        board:
+          type: string
+
+        name:
+          type: string
+
+        description:
+          type: string
+
+        console:
+          type: string
+          pattern: "^/dev/.*$"
+
+        fastboot:
+          type: string
+          pattern: "^[0-9a-f]{8}$"
+
+        fastboot_set_active:
+          type: boolean
+
+        broken_fastboot_boot:
+          type: boolean
+
+        usb_always_on:
+          type: boolean
+
+        fastboot_key_timeout:
+          type: integer
+          minimum: 1
+
+        cdba:
+          type: string
+          pattern: "^/dev/.*$"
+
+        conmux:
+          type: string
+          pattern: "^/dev/.*$"
+
+        external:
+          type: string
+
+        ppps_path:
+          type: string
+
+        ppps3_path:
+          type: string
+
+        qcomlt_debug_board:
+          type: string
+          pattern: "^/dev/.*$"
+
+        voltage:
+          type: integer
+
+        alpaca:
+          type: string
+          pattern: "^/dev/.*$"
+
+        users:
+          type: array
+          uniqueItems: true
+          minItems: 1
+          items:
+            type: string
+
+        ftdi_gpio:
+          oneOf:
+            - type: string
+              # <libftdi description>;[<interface>[;<gpios>...]]
+              # - libftdi description: ftdi_usb_open_string() formats
+              # - interface: A, B, C or D (default A)
+              # - gpios: type,id,polarity
+              #  - type: POWER, FASTBOOT_KEY, POWER_KEY or USB_DISCONNECT
+              #  - id: 0, 1, 2, 3, 4, 5, 6 or 7
+              #  - polarity: ACTIVE_HIGH or ACTIVE_LOW
+              pattern: "^(d:/.*)|(i:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}(:[0-9]+)?)|(s:0x[0-9a-fA-F]{4}:0x[0-9a-fA-F]{4}:FT[0-9a-zA-Z]{6})(;[A-D](;(POWER|FASTBOOT_KEY|POWER_KEY|USB_DISCONNECT),[0-7],(ACTIVE_HIGH|ACTIVE_LOW)){0,4})?$"
+              deprecated: true
```

Comment:
I think, let's drop the deprecated entry in schema. Let's support only the new spec.

Reply:
This description is used in production right now and is part of cdba for at least 1 year now, so dropping it would be quite brutal, and I did the regex already !!

Reply:
But we want to encourage people to migrate to proper schema... But probably deprecated is fine :-)